### PR TITLE
Add admin no trn page to mark user as not in DQT

### PR DESCRIPTION
**Pull request by wilpow-github**
### Context

As a TRA helpdesk agent
I want to record that I have searched for a user in DQT but not found a match
So that NPQ can be informed that the user does not have a TRN

### Changes proposed in this pull request

Add no TRN page. When user marked as not in DQT, update TRN lookup status

### Checklist

-   [x] Attach to Trello card
-   [ ] Rebased master
-   [ ] Cleaned commit history
-   [ ] Tested by running locally
-   [ ] Reminder created to manually clean any removed app settings post deployment

### CODE REVIEW

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/AssignTrn/NoTrn.cshtml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+@page "/admin/users/{userId}/no-trn"
+@model TeacherIdentity.AuthServer.Pages.Admin.AssignTrn.NoTrn
+@{
+    ViewBag.Title = @Html.DisplayNameFor(m => m.HasNoTrn);
+}
+
+@section BeforeContent {
+    <govuk-back-link asp-page="/Admin/User" asp-route-userId="@Model.UserId" />
```
Comment (wilpow-github) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/AssignTrn/NoTrn.cshtml:8]: Not sure if makes more sense to redirect back to /Admin/Users here

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/AssignTrn/Trn.cshtml.cs [hunk 2]
```diff
@@ -25,7 +25,7 @@ public TrnModel(TeacherIdentityServerDbContext dbContext, IDqtApiClient dqtApiCl
     [BindProperty]
     [Display(Name = "What is their teacher reference number (TRN)?")]
     [Required(ErrorMessage = "Enter a TRN")]
-    [RegularExpression(@"^\d{7}$", ErrorMessage = "TRN must be 7 digits")]
+    [RegularExpression(@"\A\D*(\d{1}\D*){7}\D*\Z", ErrorMessage = "TRN must be 7 digits")]
```
Comment (wilpow-github) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/AssignTrn/Trn.cshtml.cs:28]: Updated regex to same as in Trn/HasTrnPage

Comment (gunndabad) [reply dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/AssignTrn/Trn.cshtml.cs:28]: I think we want the simpler regex here (the support staff should know to always use a numeric form of the TRN).

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/AssignTrn/NoTrn.cshtml (hunk 1 above)
Comment (gunndabad) [reply dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/AssignTrn/NoTrn.cshtml:8]: Agreed. We can loop back and make sure we have the right links from both the users list and user detail page later on.

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/AssignTrn/Trn.cshtml.cs (hunk 2 above)
Comment (wilpow-github) [reply dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/AssignTrn/Trn.cshtml.cs:28]: Ah ok didn't realise they could always be represented as 7 digits

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/Users.cshtml [hunk 3]
```diff
@@ -20,15 +20,17 @@
...
                         <th scope="col" class="govuk-table__header"></th>
+                        <th scope="col" class="govuk-table__header"></th>
                     </tr>
                 </thead>
                 <tbody class="govuk-table__body">
                     @foreach (var user in Model.Open!)
                     {
                         <tr class="govuk-table__row" data-testid="user-@user.UserId">
-                            <td class="govuk-table__cell"><a asp-page="EditUser" asp-route-userId="@user.UserId">@user.Name</a></td>
+                            <td class="govuk-table__cell"><a asp-page="User" asp-route-userId="@user.UserId">@user.Name</a></td>
```
Comment (wilpow-github) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/Users.cshtml:30]: Another case of EditUser

**Review by gunndabad — APPROVED** [review]

**Pull request merged**